From d247be177736e74ad8d3f9148b4d1d74e9f0e030 Mon Sep 17 00:00:00 2001
From: Kihwal Lee <kihwal@apache.org>
Date: Tue, 6 Feb 2018 15:43:19 -0600
Subject: [PATCH 2774/2863] HADOOP-15212. Add independent secret manager
 method for logging expired tokens. Contributed by
 Daryn Sharp.

(cherry picked from commit 2dd960de983a30bf0d9ee957bdb09f825f9d40a3)
(cherry picked from commit 6ea2a9389e8bd1b5aa35d01c5b1556f892181f11)

 Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt

Change-Id: If4555493f0e911835abda90f4a76fcef81143ed1
---
 .../AbstractDelegationTokenSecretManager.java      |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java
index fa0b428..f408bae 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java
@@ -29,6 +29,7 @@
 import java.io.DataOutputStream;
 import java.io.IOException;
 import java.security.MessageDigest;
+import java.util.Collection;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Iterator;
@@ -620,6 +621,11 @@ private void removeExpiredToken() throws IOException {
       }
     }
     // don't hold lock on 'this' to avoid edit log updates blocking token ops
+    logExpireTokens(expiredTokens);
+  }
+
+  protected void logExpireTokens(
+      Collection<TokenIdent> expiredTokens) throws IOException {
     for (TokenIdent ident : expiredTokens) {
       logExpireToken(ident);
       LOG.info("Removing expired token " + formatTokenId(ident));
-- 
1.7.9.5

